### ISSUE TITLE
Flaky test 'InteractiveHostDesktopTests.StackOverflow'

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#46414]
https://github.com/dotnet/roslyn/blob/4d0c2c0d6fc6eb4d301454ec2e158b63b090f098/src/Interactive/HostTest/InteractiveHostDesktopTests.cs#L71

Example build: https://dev.azure.com/dnceng/public/_build/results?buildId=749582&view=ms.vss-test-web.build-test-results-tab&runId=23240534&resultId=202015&paneView=debug

Xunit log:
```
    Microsoft.CodeAnalysis.UnitTests.Interactive.InteractiveHostDesktopTests.StackOverflow [FAIL]
      Assert.Equal() Failure
                                        (pos 52)
      Expected: úúúckOverflowException.\nHosting process exited with exit code -1úúú
      Actual:   úúúckOverflowException.
                                        (pos 52)
      Stack Trace:
        /_/src/Interactive/HostTest/InteractiveHostDesktopTests.cs(94,0): at Microsoft.CodeAnalysis.UnitTests.Interactive.InteractiveHostDesktopTests.StackOverflow()
        --- End of stack trace from previous location ---
```

**Comment by RikkiGibson** [comment]
Another run containing this failure

https://dev.azure.com/dnceng/public/_build/results?buildId=778472&view=ms.vss-test-web.build-test-results-tab&runId=24376582&resultId=100019&paneView=debug

@tmat would a PR be accepted which accepts *either* of these strings or perhaps checks a prefix match in the error message?

**Comment by tmat** [comment]
Yes, accepting either of these seems a fine workaround. Let's keep an issue opened to track investigation of why are we not seeing the rest of the message. It's not critical though that we do not.

**Comment by jaredpar** [comment]
Can we please move forward with this change? This test is the #1 most flaky test in Roslyn right now. It's failing at least one build per day.
